src/kbmod/standardizers/standardizer.py

```diff
@@ -0,0 +1,591 @@
+        resolved the target are sorted based on the `Standardizer` priority and
+        used to standardize the target.
+
+        Because each `Standardizer` is asked to resolve each target, of which
+        there are potentially many, it is encouraged that this method is
+        implemented with performance in mind.
+
+        Sometimes, however, it may not be possible to avoid acquiring or
+        constructing a resource that, in the given context, could be considered
+        expensive. Returning the resource(s) allows that cost to be optimized
+        away, by avoiding acquiring or constructing the resource again during
+        initialization.
+
+        On a practical example: `FitsStandardizer` standardize FITS files
+        pointed to by a local filesystem path (i.e. the target). To confirm
+        their ability to standardize the target, they have to make sure they
+        can open and read the file. They use AstroPy to construct an
+        `~astropy.io.fits.HDUList`, i.e. the resource. Returning the already
+        constructed `~astropy.io.fits.HDUList` allows `FitsStandardizer` to
+        skip constructing a new one via `astropy.io.fits.open`.
+
+        Parameters
+        ----------
+        tgt : data-source
+            Some data source, URL, URI, POSIX filepath, Rubin Data Butler Data
+            Repository, an Python object, callable, etc.
+
+        Returns
+        -------
+        canStandardize : `bool`
+            `True` when the processor knows how to handle uploaded
+            file and `False` otherwise.
+        resources : `dict`, optional
+            Any optional resources constructed during target resolution. An
+            empty dictionary if there are none.
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def canStandardize(cls, tgt):
+        """Returns ``True`` when the standardizer knows how to process
+        the given upload (file) type.
+
+        Parameters
+        ----------
+        tgt : data-source
+            Some data source, URL, URI, POSIX filepath, Rubin Data Butler Data
+            Repository, an Python object, callable, etc.
+
+        Returns
+        -------
+        canStandardize : `bool`
+            `True` when the processor knows how to handle uploaded
+            file and `False` otherwise.
+
+        Notes
+        -----
+        Implementation is instrument-specific.
+        """
+        return cls.resolveTarget(tgt)[0]
+
+    def __init_subclass__(cls, **kwargs):
+        # Registers subclasses of this class with set `name` class
+        # parameters as availible standardizers. Note the specific
+        # standardizer has to be imported before this class since the
+        # mechanism is triggered at definition time.
+        name = getattr(cls, "name", False)
+        if name and name is not None:
+            super().__init_subclass__(**kwargs)
+            Standardizer.registry[cls.name] = cls
+
+    @abc.abstractmethod
```

Comment:
In short, I don't want to allow the three following things:
* direct instantiation of Standardizer
* enforce setting of location (which I should probably rename `tgt` or something for generality) and `config`
* enforce existence of `processable`

If I make `__init__` non-abstract then the user could inherit `Standardizer` and not overwrite the `__init__`. Then they are not forced to handle the data source IO in it (tests should hopefully enforce that it roundtrips in `ImageCollection`). If they don't do that and don't call `super().__init__(location="something")` - the location won't be set to anything meaningful. Even if it isn't meaningful - the `config` doesn't get instantiated. This is the only place that sets `self.config = self.configClass(config)` so it must execute - otherwise each `Standardizer` wouldn't have a `config` by default either. 

I guess the core of your question is "how does abstractmethod execute anything at all"? There are no short answers to that, but I'm happy to chat irl. Python's `__init__` is just one step in the construction of an instance. Before it, the `metaclass` mechanism is executed where `__new__` is called, builds an object using [descriptors](https://docs.python.org/3/howto/descriptor.html#descriptor-howto-guide) and then various hooks can be called (like the `__init__subclass__` we use). These already see fully initialized descriptors so they can do whatever they want to them. After that which the `__init__` itself it called. In this process, `__init__`, is supposed to just initialize (not so much construct) the values of descriptors. 

`Standardzier` inherits from `abc.ABC` Abstract Base Class, which does part of this heavy-lifting in the `__init__` and we leverage `__init_subclass__` to keep the `registry` of subclasses working. So that's why declaring things in the `__init__` actually executes - as long as the user follows the golden rule of invoking the `super().__init__` once they inherit. For most part the complicated IO mechanisms are hidden from them, like in the `FitsStandardizer` for example that provides a generic IO for FITS files and HDUList objects, and then provides a reasonable best-guess default behavior for `processable` in `SingleExtensionFits` and `MultiExtensionFits` classes. Inheriting from those, they would just have to worry about writing the `.standardize*` methods.